djangocms_navigation/forms.py

```diff
@@ -47,17 +69,6 @@ def __init__(self, *args, **kwargs):
             MenuItem, for_node=self.menu_root.get_root()
         )
 
-        # Todo: change when autocomplete implemented, tests for this field
-        content_choices = []
-        for model in supported_models():
-            content_choices.extend([(obj.id, obj) for obj in model.objects.all()])
-        self.fields["object_id"].widget.choices = content_choices
-
-        # TODO: If this initial still needed after autocomplete changes
-        # then don't forget to write tests for this
-        if self.instance:
-            self.fields["object_id"].initial = self.instance.object_id
-
     def clean(self):
         cleaned_data = super().clean()
 
```

Comment:
Perhaps add validation for content_type/object_id, whether there's actually an object in the db.

Reply:
This could be tricky as content_type and object_id is not necessary for root item.

Reply:
well, detect it's a root item then and skip validation